src/amqp/publisher.ts

```diff
@@ -1,25 +1,34 @@
 import amqp from 'amqplib';
 import Debug from 'debug';
 
-import { PubSubAMQPConfig, Exchange, Queue } from './interfaces';
+import { PubSubAMQPConfig, Exchange } from './interfaces';
 
 export class AMQPPublisher {
   private connection: amqp.Connection;
-  private exchange: Exchange;
+  private exchange: Exchange = {
+    options: {
+      durable: false,
+      autoDelete: false
+    }
+  };
   private channel: amqp.Channel | null = null;
 
   constructor(
-    private config: PubSubAMQPConfig,
+    config: PubSubAMQPConfig,
     private logger: Debug.IDebugger
   ) {
     this.connection = config.connection;
-    this.exchange = config.exchange;
+    this.exchange = { ...config.exchange };
```

Comment:
Doesn't seem like we should need the spread into object for all of these constructors.

Reply:
Reduced the amount of spreads. but i kept some for defaulting purposes